pydantic/dataclasses.py

```diff
@@ -211,3 +211,8 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
 
 
 __getattr__ = getattr_migration(__name__)
+
+if (3, 8) <= sys.version_info < (3, 11):
```

Comment:
Without this, `dataclasses.InitVar[<type>]` raises errors when returned by `typing.get_type_hints`:

```
>           raise TypeError(f"{msg} Got {arg!r:.100}.")
E           TypeError: Forward references must evaluate to types. Got dataclasses.InitVar[int].
```
The issue is, even if you were willing to rely on dataclasses to tell you what is or isn't an InitVar, you really need this function (or something similar) to allow you to resolve forward references to get what the generic parameter resolves to.

I think monkeypatching `InitVar` like this is preferable to copying the whole logic of `get_type_hints` with a tiny tweak to not error if the result isn't callable. But I suppose I could be convinced away from this if we think it's really problematic.